javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -165,8 +164,8 @@ private Vector(BitMappedTrie<T> trie) {
             final Object[] array = ((Collection<? extends T>) iterable).toArray();
             return ofAll(BitMappedTrie.ofAll(array));
         } else {
-            final Seq<? extends T> elems = seq(iterable);
-            final Object[] array = asArray(elems.iterator(), elems.size());
+            final Seq<? extends T> result = (iterable instanceof Seq) ? (Seq<T>) iterable : List.ofAll(iterable);
+            final Object[] array = asArray(result.iterator(), result.size());
```

Comment:
I thought about it, but if the `iterable` is `TraversableOnce`, I would consume it for the `size`, right?